src/components/FileUpload/FileUpload.stories.jsx

```
@@ -8,13 +8,45 @@ export default {
 };
 
 const mockCreateUploadSuccess = () => {
-  return Promise.resolve();
+  return Promise.resolve({ id: '1234' });
 };
 
 const mockCreateUploadError = () => {
   return Promise.reject();
 };
+const Template = (args) => <FileUpload {...args} />;
 
-export const fileUploadSuccess = () => <FileUpload createUpload={mockCreateUploadSuccess} />;
+export const FileUploadSuccess = Template.bind({});
+FileUploadSuccess.args = {
+  createUpload: mockCreateUploadSuccess,
+};
+
+export const FileUploadWithExtendedAcceptedFileTypes = Template.bind({});
+FileUploadWithExtendedAcceptedFileTypes.args = {
+  createUpload: mockCreateUploadSuccess,
+  acceptedFileTypes: [
+    'image/jpeg',
+    'image/png',
+    'application/pdf',
+    '.csv',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'application/vnd.ms-excel',
+  ],
+};
+
+export const FileUploadWithLimitOneFile = Template.bind({});
+FileUploadWithLimitOneFile.args = {
+  createUpload: mockCreateUploadSuccess,
+  allowMultiple: false,
+};
+
+export const FileUploadWithNoParallelUploads = Template.bind({});
+FileUploadWithNoParallelUploads.args = {
+  createUpload: mockCreateUploadSuccess,
+  maxParallelUploads: 1,
+};
```

I didn't have any luck with using the maxParallelUploads argument. 

In the ideal scenario we would want the first `createUpload` make an API call to create a new document ID and then have state be updated to capture this change. We'd want the second instance of `createUpload` call to then based on state determine if we have a document ID and then do some other logic.

More generically we want to do something the first time and then update state and then do something different the second time. 

Even when I change `maxParallelUploads` to 1 I am not able to achieve this. Have you been able to get this working?

---

I see what you're saying when this is tested on the upload orders form, it seems to queue up the second file but eventually call remove all files after the first succeeds.  It could be a bug in our implementation of the callbacks as well, if we've never tried using it a different way.

I don't like some of the silent errors or warnings in the component like if you disallowed multiple and you dragged 2 files it will drop one without explanation.